### PR TITLE
QW-2404: Fix merge CLI command.

### PR DESCRIPTION
**Pull request by ChillFish8**
### Description
Fixes how we run the merge CLI command to use the new ability to detach the merge pipeline itself.

We then check the number of ongoing merges and wait until that number has reached `0` or the actor has exited.

### How was this PR tested?

Describe how you tested this PR.

### CODE REVIEW

File: quickwit/quickwit-cli/src/index.rs [hunk 1]
```diff
@@ -1069,12 +1069,33 @@ pub async fn merge_cli(args: MergeArgs) -> anyhow::Result<()> {
...
         .await?;
-    let (pipeline_exit_status, _pipeline_statistics) = pipeline_handle.join().await;
+
+    let mut check_interval = tokio::time::interval(Duration::from_secs(1));
+    loop {
+        check_interval.tick().await;
+
+        let observation = pipeline_handle.observe().await;
+
+        if observation.num_ongoing_merges == 0 {
```
Comment (fmassot) [inline quickwit/quickwit-cli/src/index.rs:1084]: Nitpick: we can add an `info` log to give the reason why the merge stopped.

File: quickwit/quickwit-cli/src/index.rs [hunk 2]
```diff
@@ -1069,12 +1069,33 @@ pub async fn merge_cli(args: MergeArgs) -> anyhow::Result<()> {
...
+            break;
+        }
+
+        if observation.obs_type == ObservationType::PostMortem {
```
Comment (fmassot) [inline quickwit/quickwit-cli/src/index.rs:1088]: Nitpick: same here.

**Review by fmassot — APPROVED** [review]
LGTM!

File: quickwit/quickwit-cli/src/index.rs (hunk 1 above)
Comment (ChillFish8) [reply quickwit/quickwit-cli/src/index.rs:1084]: Added!

File: quickwit/quickwit-cli/src/index.rs (hunk 2 above)
Comment (ChillFish8) [reply quickwit/quickwit-cli/src/index.rs:1089]: Added!

**Pull request merged**